examples/simple-generator/src/app/service.ts

```diff
@@ -39,6 +48,54 @@ export async function getServiceInfo(generator: Generator): Promise<ServiceInfo>
     };
 }
 
+/**
+ * Check the secure storage if it has credentials for teh entered url.
+ *
+ * @param url target system url
+ * @param client optional sap-client parameter
+ * @returns credentials or undefined
+ */
+async function getCredentials(url: string, client?: string): Promise<AxiosBasicCredentials | undefined> {
+    const systemService = await getService<BackendSystem, BackendSystemKey>({ entityName: 'system' });
+    const system = await systemService.read(new BackendSystemKey({ url, client }));
+    return system?.username ? { username: system.username, password: system.password! } : undefined;
+}
+
+/**
+ * Ask the user whether the credentials should be stored. If yes, store them in the secure storage.
+ *
+ * @param generator generator reference used for prompting
+ * @param provider service provider for which the credentials should be stored
+ */
+async function storeCredentials(generator: Generator, provider: AbapServiceProvider) {
+    const { storeCreds, name }: { storeCreds: boolean; name: string } = await generator.prompt([
+        {
+            type: 'confirm',
+            name: 'storeCreds',
+            message: 'Do you want to store your credentials in the secure storage?',
+            default: true
+        },
+        {
+            type: 'input',
+            name: 'name',
+            message: 'System name:',
+            default: new URL(provider.defaults.baseURL!).hostname,
+            validate: (answer) => !!answer
+        }
+    ]);
+    if (storeCreds) {
+        const systemService = await getService<BackendSystem, string>({ entityName: 'system' });
+        const system = new BackendSystem({
+            name,
+            url: provider.defaults.baseURL!,
+            client: provider.defaults.params?.['sap-client'],
+            username: provider.defaults.auth?.username,
+            password: provider.defaults.auth?.password
+        });
+        await systemService.write(system);
```

Comment:
If for some reason the ```write``` fails, will the store return an error or should we handle it here as well?

Reply:
I am not fully sure to be honest, however, you make a very good point here, we need to handle the possibility that something goes wrong. Storing the credentials is not critical for the generation process, therefore, I wrapped it in a `trycatch` and log the error and continue. https://github.com/SAP/open-ux-tools/pull/649/commits/7518ecc7b8cdfcf579312c8da1734ef823abcf73
Ok for you?

Reply:
👍